conn.go

```diff
@@ -347,6 +376,97 @@ func (conn *Conn) inWorker() {
 	}
 }
 
+// sigWorker runs in an own goroutine, getting signals and handling
+// them appropriately.
+func (conn *Conn) sigWorker() {
+WorkerLoop:
+	for {
+		signal := conn.nextSignal()
+		if signal != nil {
+			closing := conn.handleSignal(signal)
+			if closing {
+				break WorkerLoop
+			}
+		} else {
+			select {
+			case <-conn.signalQueueWakeUp:
+				continue WorkerLoop
+			case <-conn.shutdownSigWorker:
+				break WorkerLoop
+			}
+		}
+	}
+}
+
+// Gets next signal from queue or nil if there was no signal in queue.
+func (conn *Conn) nextSignal() *Signal {
+	conn.signalQueueLck.Lock()
+	defer conn.signalQueueLck.Unlock()
+	for len(conn.signalQueue) > 0 {
+		signal := conn.signalQueue[0]
+		conn.signalQueue = conn.signalQueue[1:]
+		if signal != nil {
+			return signal
+		}
+	}
+
+	conn.doWakeUp = true
+	return nil
+}
+
+// Either adds new pending channels to main signal channels array or
+// removes them if they already existed there.
+//
+// It is assumed that signals lock is locked.
```

Comment:
It would help to be explicit here and refer to the actual variable name rather than just "signals lock"